python/tvm/relay/op/contrib/cmsisnn.py

```diff
@@ -52,37 +54,85 @@ def partition_for_cmsisnn(mod, params=None, **opts):
             transform.InferType(),
             transform.MergeComposite(pattern_table()),
             transform.AnnotateTarget("cmsisnn"),
-            transform.MergeCompilerRegions(),
```

Comment:
Why are we not running this ?

Reply:
Although, enabling this has no impact on the generated primfunc, but wanted to keep it simple to have 1-1 mapping of single external Relay function to single primfunc.

Reply:
The way the current BYOC works, this will create a source per operator and also additional set of indirection will be there for each call.

i.e.
main_func <--> ext_func <--> cmsis_func
main_func <--> ext_func <--> cmsis_func
main_func <--> ext_func <--> cmsis_func

vs

main_func <--> ext_func <--> cmsis_func1, cmsis_func2, cmsis_func3

Reply:
Is there a problem with the first representation?

Reply:
Code size ? and more translation units produced ? each ext function will become its own .c file right ?

Reply:
Enabling this pass gets us the latter one. Irrespective of the Relay that we get post partitioning, RelayToTIR works on composite functions each one of which is translated into a separate primfunc. So, independent of the pass, atm, we end up producing multiple '.c'.

We made the choice of having separate primfunc per operator to get away with the bookkeeping required in RelayToTIR at the cost of having to invoke different C functions at runtime.

@Mousius you want to add something else?

Reply:
With https://github.com/apache/tvm/pull/9397 the `Target` will produce a single `IRModule` into `runtime::Module` rather than creating many so the functions are just wrappers around the CMSIS-NN calls after that which essentially create the CMSIS-NN structures and call them so that seems the minimal amount for them other than packing multiple CMSIS-NN API calls into a single PrimFunc which we didn't intend to do.

Reply:
Thanks! @Mousius. Cool as long as there is not overhead of in-direction and multiple source files -- we should be good. :)